### DISCUSSION TITLE
No android emulators for Flutter integration tests

### DISCUSSION THREAD

**Discussion by charbel-elhajj** [codemagic-ci-cd/codemagic-docs#1216] [Q&A] [answered]
Hello everyone.

I am trying to run integration tests on my app in my codemagic pipeline. I would like to run them on both Android and iOS.
However, it seems like the android option is not available. (see screenshot below) 

![Screenshot 2022-12-07 at 09 41 25](https://user-images.githubusercontent.com/72244696/206130885-7935e5f4-ce95-4f6f-99b1-223b64dd4de2.png)

I also tried to launch the emulator manually using `flutter emulators --create --name emulator` but it resulted in the following error:

```
Failed to create emulator 'emulator'.

No suitable Android AVD system images are available. You may need to install these using sdkmanager, for example:
  sdkmanager "system-images;android-27;google_apis_playstore;x86"

You can find more information on managing emulators at the links below:
  https://developer.android.com/studio/run/managing-avds
  https://developer.android.com/studio/command-line/avdmanager
```

**Comment by himesh-cm** [comment, marked as answer]
Hi @charbel-elhajj, if you are using M1 machines, then android emulators are not supported. You can switch to Linux or Mac pro instances.

**Reply by charbel-elhajj** [reply]
Hello @himesh-cm, it solved it! Thank you

**Comment by pfaffenrodt** [comment]
Mac Pro instance "The mac_pro instance type does not support the selected Xcode version. Please consider using mac_mini_m1 instead"

Mac Pro seems to be deprecated. There should be emulators available for mac mini. Crossbuild/tests are no longer possible.

**Reply by himesh-cm** [reply]
Hi @pfaffenrodt, You can still use Mac pro machines with 14.2 or build using Linux machines.

**Reply by pfaffenrodt** [reply]
I know, that I can. But you can not build crossbuild on the same maschine. For example in Kmm or Flutter projects with latest xcode.
